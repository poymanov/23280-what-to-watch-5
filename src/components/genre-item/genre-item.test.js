import React from "react";
import renderer from "react-test-renderer";
import GenreItem from "./genre-item";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../etc/browser-history";

const noop = () => {};

const genre = {
  id: `all`,
  title: `All`
};

describe(`Should GenreItem render correctly`, () => {
  it(`Not active item`, () => {
    const tree = renderer
      .create(
          <BrowserRouter history={browserHistory}>
            <GenreItem genre={genre} isActive={false} handleGenreClick={noop} />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Active item`, () => {
    const tree = renderer
      .create(
          <BrowserRouter history={browserHistory}>
            <GenreItem genre={genre} isActive={true} handleGenreClick={noop} />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
