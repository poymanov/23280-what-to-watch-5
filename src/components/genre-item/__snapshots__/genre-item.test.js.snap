// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenreItem render correctly Active item 1`] = `
<li
  className="catalog__genres-item catalog__genres-item--active"
>
  <a
    className="catalog__genres-link"
    href="/"
    onClick={[Function]}
  >
    All
  </a>
</li>
`;

exports[`Should GenreItem render correctly Not active item 1`] = `
<li
  className="catalog__genres-item "
>
  <a
    className="catalog__genres-link"
    href="/"
    onClick={[Function]}
  >
    All
  </a>
</li>
`;
